jib-plugins-common/src/main/java/com/google/cloud/tools/jib/plugins/common/UpdateChecker.java

```diff
@@ -0,0 +1,210 @@
+          Instant modifiedTime = Instant.parse(fileContents);
+          if (modifiedTime.plus(Duration.ofDays(1)).isAfter(Instant.now())) {
+            return Optional.empty();
+          }
+        } catch (DateTimeParseException | IOException ex) {
+          // If reading update time failed, file might be corrupt, so delete it
+          log.accept(LogEvent.debug("Failed to read lastUpdateCheck; " + ex.getMessage()));
+          Files.delete(lastUpdateCheck);
+        }
+      }
+
+      // Check for update
+      HttpURLConnection connection = (HttpURLConnection) new URL(versionUrl).openConnection();
+      try {
+        connection.setConnectTimeout(3000);
+        BufferedReader bufferedReader =
+            new BufferedReader(
+                new InputStreamReader(connection.getInputStream(), StandardCharsets.UTF_8));
+        VersionJsonTemplate version =
+            JsonTemplateMapper.readJson(
+                bufferedReader.readLine().trim(), VersionJsonTemplate.class);
+        Files.write(lastUpdateCheck, Instant.now().toString().getBytes(StandardCharsets.UTF_8));
+        if (currentVersion.equals(version.latest)) {
+          return Optional.empty();
+        }
+        return Optional.of(
+            "A new version of Jib ("
+                + version.latest
+                + ") is available (currently using "
+                + currentVersion
+                + "). Update your build configuration to use the latest features and fixes!");
+
+      } finally {
+        connection.disconnect();
+      }
+
+    } catch (IOException ex) {
+      log.accept(LogEvent.debug("Update check failed; " + ex.getMessage()));
+    }
+
+    return Optional.empty();
+  }
+
+  /**
+   * Returns a message indicating Jib should be upgraded if the check succeeded and the current
+   * version is outdated, or returns {@code Optional.empty()} if the check was interrupted or did
+   * not determine that a later version was available.
+   *
+   * @param updateMessageFuture the {@link Future} returned by {@link UpdateChecker#checkForUpdate}
+   * @return the {@link Optional} message to upgrade Jib if a later version was found, else {@code
+   *     Optional.empty()}.
+   */
+  public static Optional<String> finishUpdateCheck(Future<Optional<String>> updateMessageFuture) {
+    if (updateMessageFuture.isDone()) {
+      try {
+        return updateMessageFuture.get();
+      } catch (InterruptedException | ExecutionException ignored) {
+        // Fail silently;
+      }
+    }
+    updateMessageFuture.cancel(true);
+    return Optional.empty();
+  }
+
+  /**
+   * Returns the config directory set by {@link PropertyNames#CONFIG_DIRECTORY} if not null,
+   * otherwise returns the default config directory.
+   *
+   * @return the config directory set by {@link PropertyNames#CONFIG_DIRECTORY} if not null,
+   *     otherwise returns the default config directory.
+   */
+  private static Path getConfigDir() {
```

Comment:
I thought we had some sort of static class do this?

Reply:
Oh wait sorry, this uses system property as part of the plugin frontend, so it doesn't belong in jib core.